### PR TITLE
feat(Rln-Relay): Consolidates the input paths of rln credential, chat2 reads rln credential from file only

### PR DESCRIPTION
**Pull request by staheri14**
This aim of this PR is to address the issues happened during the rln-relay testnet1 regarding supplying RLN credentials. More details are in https://github.com/status-im/nwaku/issues/1069. 
With the changes included in this PR, there will be only one way to pass Rln credential, that is through `rlnCredetnial.txt` file (which hopefully soon be replaced with keystone with proper encryption)
For this sake, a new config option is made available named `rlnRelayCredPath` which is the path to the `rlnCredetnial.txt` file, if this option is passed, then the credentials are generated and persisted in the file (or retrieved from the file), otherwise, new credentials will be generated for each run but will not be persisted. 

The tutorial is also updated to reflect this change.

Note that the RLN credential config options i.e., `rln-relay-membership-index`, `rlnRelayIdCommitmentKey` and `rlnRelayIdKey` are still available (but not recommended in the chat2 tutorial) as we may find use cases for them in the future.

### CODE REVIEW

**Review by jm-clius — APPROVED** [review]
LGTM

**Comment by staheri14** [comment]
> LGTM! However, why you allow chosing the file path but not the filename?
> 
> This will address (or at least help in addressing) both
> 
> > Allow re-registration even when the rlncredentials.txt file is present. Maybe via an extra parameter --force-registration=true
> > Allowing multiple rlnCredentials.txt files for distinct chat2 apps running on the same machine. Potentially prefixing the filenames with the Ethereum account address (good but not sufficient)
> 
> by simply passing:
> 
> * a new file name to trigger a new credential registration (if file doesn't exists -> lock it -> registration -> write -> release);
> * an existing filename to load an already generated credential (if file exists -> try load);

This would also be possible by changing the path parameter, the file will always have the same name but the enclosing folder can change. Thus, there can be multiple credentials living in different paths.

File: docs/tutorial/onchain-rln-relay-chat2.md [hunk 1]
```diff
@@ -150,19 +150,15 @@ Once you are done with the test, make sure you close all the chat2 clients by ty
...
-For instance, if the previously generated credential are
-```
-your membership index is: xx
-your rln identity key is: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
-your rln identity commitment key is: 6c6598126ba10d1b70100893b76d7f8d7343eeb8f5ecfd48371b421c5aa6f012
-```
-Then, the execution command will look like this (inspect the last three config options):
-```
-./build/chat2 --fleet:test --content-topic:/toy-chat/2/luzhou/proto --rln-relay:true --rln-relay-dynamic:true --eth-mem-contract-address:0x4252105670fe33d2947e8ead304969849e64f2a6 --eth-account-address:your_eth_account --eth-account-privatekey:your_eth_private_key --eth-client-address:your_goerli_node --ports-shift=1 --rln-relay-membership-index:63 --rln-relay-id:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx --rln-relay-id-commitment:6c6598126ba10d1b70100893b76d7f8d7343eeb8f5ecfd48371b421c5aa6f012
+You may pass the `rln-relay-cred-path` config option to specify a path for 1) persisting RLN credential 2) retrieveing persisted RLN credential.
```
[the remark this answers is not in the document]
Comment (staheri14) [reply docs/tutorial/onchain-rln-relay-chat2.md]: Fixed in https://github.com/status-im/nwaku/pull/1080/commits/8d6b57419c3e33f6d56e1eb2115153df909233a7

**Comment by staheri14** [comment]
One additional point about not passing filename is that, this way we will have more control on the file format e.g., being a key-store, db,  txt, binary, etc. By limiting it to the path, we make the file format transparent to the user.

**Pull request merged**